frontend/src/app/account/profile/profile.component.ts

```diff
@@ -0,0 +1,241 @@
+        private messageService: MessageService,
+        private router: Router,
+        public formUtils: FormUtilsService,
+    ) {}
+
+    editProfileForm: FormGroup = new FormGroup({
+        username: new FormControl('', [Validators.required, invalidUsernameValidator()]),
+        email: new FormControl('', [Validators.required, Validators.email]),
+        password: new FormControl(''),
+    });
+
+    editPasswordForm: FormGroup = new FormGroup(
+        {
+            oldPassword: new FormControl(''),
+            password: new FormControl('', [
+                Validators.required,
+                invalidPasswordValidator(),
+                lowercasePasswordValidator(),
+                uppercasePasswordValidator(),
+                numericPasswordValidator(),
+                specialPasswordValidator(),
+                shortPasswordValidator(),
+                weakPasswordValidator(),
+            ]),
+            confirmPassword: new FormControl('', [Validators.required]),
+        },
+        {
+            validators: mismatchPasswordValidator('password', 'confirmPassword'),
+        },
+    );
+
+    passwordRequirements = [
+        { msg: 'At least one lowercase', check: () => this.formUtils.passwordHasNoLowercase(this.editPasswordForm) },
+        { msg: 'At least one uppercase', check: () => this.formUtils.passwordHasNoUppercase(this.editPasswordForm) },
+        { msg: 'At least one numeric', check: () => this.formUtils.passwordHasNoNumeric(this.editPasswordForm) },
+        {
+            msg: 'At least one special character',
+            check: () => this.formUtils.passwordHasNoSpecial(this.editPasswordForm),
+        },
+        { msg: 'Minimum 8 characters', check: () => this.formUtils.isPasswordShort(this.editPasswordForm) },
+    ];
+
+    ngOnInit() {
+        this.user = this.authenticationService.userValue;
+        this.showEditProfile = false;
+        this.showEditPassword = false;
+        this.isProcessingEdit = false;
+        this.isProcessingPassword = false;
+    }
+
+    onUpdateProfile() {
+        this.showEditProfile = true;
+        this.showEditPassword = false;
+    }
+
+    onUpdatePassword() {
+        this.showEditPassword = true;
+        this.showEditProfile = false;
+    }
+
+    onEditSubmit() {
+        if (this.editProfileForm.valid) {
+            this.isProcessingEdit = true;
+
+            // Check if password is correct first
+            this.authenticationService
+                .login(this.user!.username, this.editProfileForm.get('password')?.value)
+                .subscribe({
+                    next: () => {
+                        this.authenticationService
+                            .updateAccount(
+                                this.editProfileForm.get('username')?.value,
+                                this.editProfileForm.get('email')?.value,
+                                this.editProfileForm.get('password')?.value,
+                            )
```

Comment:
This should be replaced with updating the backend routes to enforce password verification